### DISCUSSION TITLE
how to add role at the time (php artisan make:filament-user)

### DISCUSSION THREAD

**Discussion by hagiik** [filamentphp/filament#15127] [Help] [answered]
I want to ask how to add a role when starting filament, the default on filament-user only displays my Name, Email, and password. so I want to add a role when adding a role using the terminal.

![image](https://github.com/user-attachments/assets/f7bcdc41-e338-4e67-ae0c-e4189a4ca6b0)

I want to add role under password

**Comment by leandrocfe** [comment]
Maybe you could create a new command extending [this](https://github.com/filamentphp/filament/blob/3.x/packages/panels/src/Commands/MakeUserCommand.php) and customizing what you need

**Reply by hagiik** [reply]
where is it located in the text editor, I can't find it because there are so many folders

**Reply by leandrocfe** [reply, marked as answer]
```bash
php artisan make:command CustomUserCommand
```

app/Console/Commands/CustomUserCommand.php
```php
namespace App\Console\Commands;

use Filament\Commands\MakeUserCommand;

class CustomUserCommand extends MakeUserCommand
{
    protected function getUserData(): array
    {
        return [
            ...parent::getUserData(),
            'role' => $this->choice('Role', ['admin', 'user'], 0),
        ];
    }
}

```

**Reply by hagiik** [reply]
it works, I just have to connect it to the spatie. 

thank you very much in advance 👍
